File: framework/src/sbt-plugin/src/sbt-test/routes-compiler-plugin/injected-routes-compilation/app/controllers/InstanceController.scala
/*
 * Copyright (C) 2009-2016 Lightbend Inc. <https://www.lightbend.com>
 */
package controllers

import play.api.mvc._

class InstanceController extends Controller {
  var invoked = 0

  def index = Action {
    invoked += 1
    Ok(invoked.toString)
  }
}

File: dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/routes-compiler-injected-routes-compilation-with-request-passed/app/controllers/InstanceController.java
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package controllers;

import play.mvc.*;

public class InstanceController extends Controller {
  private int invoked = 0;

  public Result index(Http.Request req) {
    this.invoked += 1;
    return ok(req.uri() + " " + invoked);
  }
}
